client/cloudevents/protocol/protocol.go

```diff
@@ -0,0 +1,22 @@
+package protocol
+
+import (
+	"context"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"open-cluster-management.io/api/client/cloudevents/protocol/mqtt"
+)
+
+// CloudEventsClient is used to send/receive cloudevents. Each event protocol has its own implementation.
+//
+// Available implementations:
+//   - MQTT
+type CloudEventsClient interface {
```

Comment:
I am thinking the interface should be like

```
type CloudEventsClient interface {
    Sender() cloudevents.Client
    Subscriber() cloudevents.Client
    WithContext(ctx context.Context, opts) context.Context
}
```

and the mqtt.options returns the interface with

options.Source()
options.Agent()

Reply:
Correct! 
- nit:
  ```go
  Sender -> Receiver
  Publisher -> Subscriber
  ```

- and also about the WithContext, Is it possible to consider accepting a dynamic parameter? like
  ```go
  WithContext(ctx context.Context, opts ...interface{}) context.Context
  ```

Reply:
Use `CloudEventsSourceOptions` and `CloudEventsAgentOptions` instead of this interface when building a coudeventsclient